SonarQube.TeamBuild.PreProcessor/Interfaces/IAnalyzerProvider.cs

```diff
@@ -19,6 +20,6 @@ public interface IAnalyzerProvider
         /// </summary>
         /// <param name="projectKey">Identifier for the project being analyzed</param>
         /// <returns>The settings required to configure the build for Roslyn analysis</returns>
-        AnalyzerSettings SetupAnalyzers(ISonarQubeServer server, TeamBuildSettings settings, string projectKey);
+        AnalyzerSettings SetupAnalyzers(ISonarQubeServer server, TeamBuildSettings settings, string projectKey, string projectBranch);
```

Comment:
Every piece of code that previously passed projectKey now needs to also pass projectBranch, but ultimately this is just to pass it on down the call stack until we reach the ISonarQubeServer where it is composed.